include/gridtools/common/tuple_util.hpp

```diff
@@ -411,8 +414,58 @@ namespace gridtools {
             using group_indices = meta::group<group_predicate_proxy_f<Pred>::template apply,
                 extract_indices,
                 meta::zip<Types, meta::make_indices_for<Types>>>;
+
+            template <class...>
+            struct is_constructible_from_elements : std::false_type {};
+
+            template <class T, template <class...> class L, class... Ts>
+            struct is_constructible_from_elements<T, L<Ts...>> : std::is_same<decltype(T{std::declval<Ts>()...}), T> {};
+
+            template <class...>
+            struct is_getter_valid;
+
+            template <class T, class Getter, class Types, class I>
+            struct is_getter_valid<T, Getter, Types, I> : bool_constant<
+                std::is_same<decltype(Getter::template get<I::value>(std::declval<T const&>())), meta::at<Types, I> const &>::value &&
+                std::is_same<decltype(Getter::template get<I::value>(std::declval<T&>())), meta::at<Types, I>&>::value &&
+                std::is_same<decltype(Getter::template get<I::value>(std::declval<T&&>())), meta::at<Types, I>&&>::value
+            > {};
+
+            template <class T,
+                class Types = traits::to_types<T>,
+                class FromTypes = traits::from_types<T>,
+                class Getter = traits::getter<T>>
+            struct is_tuple_like : bool_constant<
+                meta::is_list<Types>::value &&
+                meta::all_of<meta::not_<std::is_void>::apply, Types>::value &&
+                std::is_same<meta::rename<FromTypes::template apply, Types>, T>::value &&
```

Comment:
what's this about?

Reply:
Can you add to the description at the top of the file of `tuple_to_types` and `tuple_from_types` an example for a non trivial case?

Reply:
I added more comments.
I am not sure what non trivial case you have in mind.
Can you write github suggestion?

Reply:
`to_types<T>` for all the normal cases will probably return just itself (`T`). My question was, what's the example where it does not. But in the meantime I answered myself: `array`.